
bswapXX collides with glibc definitions, which uses __bswap_XX instead of
__bswapXX.

--- a/sys/sys/endian.h
+++ b/sys/sys/endian.h
@@ -53,6 +53,10 @@
 #define	_UINT64_T_DECLARED
 #endif
  
+#if !defined(_KERNEL) && defined(__GLIBC__)
+/* Avoid conflicting (but functionally equivalent) definitions below. */
+#include <endian.h>
+#else
 /*
  * General byte order swapping functions.
  */
@@ -93,6 +97,7 @@
 #define	le32toh(x)	bswap32((x))
 #define	le64toh(x)	bswap64((x))
 #endif /* _BYTE_ORDER == _LITTLE_ENDIAN */
+#endif
 
 /* Alignment-agnostic encode/decode bytestream to/from little/big endian. */
 
--- a/sys/amd64/include/endian.h
+++ b/sys/amd64/include/endian.h
@@ -60,7 +60,9 @@
  * Deprecated variants that don't have enough underscores to be useful in more
  * strict namespaces.
  */
-#if __BSD_VISIBLE
+#if 1
+#include <endian.h>
+#else
 #define	LITTLE_ENDIAN	_LITTLE_ENDIAN
 #define	BIG_ENDIAN	_BIG_ENDIAN
 #define	PDP_ENDIAN	_PDP_ENDIAN
--- a/sys/i386/include/endian.h
+++ b/sys/i386/include/endian.h
@@ -60,7 +60,9 @@
  * Deprecated variants that don't have enough underscores to be useful in more
  * strict namespaces.
  */
-#if __BSD_VISIBLE
+#if 1
+#include <endian.h>
+#else
 #define	LITTLE_ENDIAN	_LITTLE_ENDIAN
 #define	BIG_ENDIAN	_BIG_ENDIAN
 #define	PDP_ENDIAN	_PDP_ENDIAN
